### PR TITLE
 queryset: add aupdate (in-place codegen transform)

### PR DESCRIPTION
**Pull request by bufke**
Adds `aupdate` by **transforming Django's `QuerySet.update()` in place** — no hand-copied function body. Builds on @NorthIsUp's NorthIsUp/django-async-backend#3 (the `async_connections` compiler routing, the `_update` returning-fields await fix, and the test are from #3), but generates `aupdate` from Django's source rather than freezing a copy of it.

### What the diff contains

| Part | Lines | Hand (claude)-written? |
|---|---|---|
| `codemon/utils.py` | **1** (`asname: str` → `str \| None`) | Yes — the *only* transformer-code change. No `__main__.py` change. |
| `codemon/config/db__models__query.yaml` | +33/−7 | Yes — declarative recipe + 2 new imports + preparatory `_insert`/`_update`/`_raw_delete` routing |
| `django_async_backend/db/models/query.py` | +60/−8 | **No** — regenerated by codemon; committed because the project commits generated files |
| `tests/…/test_aupdate.py` | +45 | Yes — from #3 |
| `README.md` | 1 | Yes |

So: the **transformer-code** change is a single line (the `With` matcher already had the asname-less fallback; only the pydantic model forbade it). Everything else is declarative config, regenerated output, and tests — there is no hand-written async function body anywhere.

### How `aupdate` is generated

`update` → `aupdate` via config: `to_async` + `rename` + `context_managers(to_async)` (flips `with` → `async with`) + `calls` (rename `transaction.mark_for_rollback_on_error` to its async form, route `get_compiler` through `async_connections`, `await execute_sql`). The `order_by` inlining and the rest of `update()`'s body regenerate from Django each release (the `order_by` loop correctly stays sync — it iterates a tuple). `sql/query.py` and `sql/compiler.py` regenerate byte-identical to `main`. Tests 5/5, full query package green.

### Scope note

`aupdate` does **not** call `_insert`/`_update`/`_raw_delete`. The `get_compiler`-routing additions to those three (and the `_update` returning-fields await — a real but currently *unreachable* latent fix, no async caller until `acreate`/`adelete` land) are preparatory groundwork carried from #3, not exercised by this PR's tests. Kept here for continuity with #3; happy to split them out if you'd prefer this PR strictly minimal.

### Known boundary — input wanted

`codemon`'s `rename` only renames the `FunctionDef`, not Django's trailing `update.alters_data = True` / `update.queryset_only = False` class-body assignments. They reference the now-renamed function, so the existing `assigns: remove` config deletes them — generated `aupdate` lacks those markers (#3's copy re-added them by hand). Minor practical impact (`.alters_data` blocks template invocation + signals queryset/manager copying), but it's the precise edge where peephole transforms stop. An `assigns: rename` primitive would close it — happy to follow up if you agree with the direction.

Credit: @NorthIsUp (NorthIsUp/django-async-backend#3).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add an async `aupdate` QuerySet method generated from Django's synchronous `update`, plus supporting async compiler routing and tests.

New Features:
- Introduce an `aupdate` async QuerySet method that mirrors Django's synchronous `update` behavior and returns the affected row count.

Enhancements:
- Route internal async `_update`, `_insert`, and `_raw_delete` operations through async database connections and await their compiler calls.
- Relax codemon context manager config to support async transforms without requiring an `as` target name.
- Document `aupdate` availability in the async backend feature matrix.

Tests:
- Add async tests covering `aupdate` behavior for bulk updates, filtered updates, F-expression updates, zero-match behavior, and sliced-query errors.

### CODE REVIEW

**Comment by bufke** [comment]
@sourcery-ai review

**Comment by Arfey** [comment]
@bufke hey, good to see you 😌

Looks good at a first glance. I'll take a closer look during the week.

File: codemon/config/db__models__query.yaml [hunk 1]
```diff
@@ -353,13 +355,25 @@ module:
...
+          context_managers:
+            - to_async: true
+          calls:
+            - func:
+                value: transaction
+                attr: mark_for_rollback_on_error
+              replace_raw: 'async_mark_for_rollback_on_error(using=self.db)'
+            - func:
+                attr: get_compiler
+              replace_raw: 'query.get_compiler(connection=async_connections[self.db])'
```
Comment (Arfey) [inline codemon/config/db__models__query.yaml]: Passing connection as a parameter sidesteps introducing `UpdateQuery` and `SQLUpdateCompiler`, but both are needed for correct ORM behavior:

- SQLUpdateCompiler handles related updates.
- UpdateQuery powers update_batch, which is used during deletes.

Both should be added to keep the behavior fully compatible

**Comment by Arfey** [comment]
@bufke ping me when it's ready 😌

File: codemon/config/db__models__query.yaml (hunk 1 above)
Comment (bufke) [reply codemon/config/db__models__query.yaml]: - Introduced async_sql.UpdateQuery in django_async_backend/db/models/sql/subqueries.py, generated by codemon.
- aupdate and _update now do self.query.chain(async_sql.UpdateQuery) instead of using the stock class. Fixes:
    - SQLUpdateCompiler.execute_sql walks self.query.get_related_updates() and constructs new query instances. With the chained async class, those instances are also async_sql.UpdateQuery, so their get_compiler(using) routes through async_connections and the await ... execute_sql(...) line works. Previously this branch would have tried to await a sync int - latent but reachable on parent-field updates across inherited models.
    - update_batch is now async. Not user-callable today, but it's the entry point Collector.delete cascades use, so it's ready for when adelete lands.

Scope deliberately kept narrow
- Only UpdateQuery.update_batch is transformed. DeleteQuery / InsertQuery stay sync. Async versions can land later The subqueries module is there to make those follow-ups small.

Also addressed
- aupdate.alters_data = True was being dropped by codemon's assigns: remove. Added a rename primitive so the attribute is rewritten from update.alters_data to aupdate.alters_data rather than removed. Sourcery flagged.

One thing I noticed but didn't fix
- SQLUpdateCompiler.pre_sql_setup at compiler.py:2384 has a sync for rows in query.get_compiler(...).execute_sql(MULTI): that iterates a coroutine. Already marked ## todo: fix in the compiler yaml.

**Comment by bufke** [comment]
Ready for review

File: codemon/config/db__models__query.yaml (hunk 1 above)
Comment (Arfey) [reply codemon/config/db__models__query.yaml]: We need an async `SQLUpdateCompiler` so we can revert changes like:

```python
query.get_compiler(connection=async_connections[self.db])
```

back to:

```python
query.get_compiler(self.db)
```

and stop relying on the sync compiler 🙂

Comment (bufke) [reply codemon/config/db__models__query.yaml]: Changes made, took me some time to walk through sql/compiler.py. Both places where query chains through async_sql.UpdateQuery (aupdate and _update) now use the native query.get_compiler(self.db) and route through AsyncDatabaseOperations.compiler_module to the async SQLUpdateCompiler. No more connection=async_connections.

I left _raw_delete and _insert on the explicit-connection form for now: they still chain through Django's sync sql.DeleteQuery / sql.InsertQuery. Can address that later.

File: django_async_backend/db/models/query.py [hunk 2]
```diff
@@ -786,10 +788,58 @@ async def _raw_delete(self, using):
...
+                new_order_by.append(annotation)
+            else:
+                new_order_by.append(col)
+        query.order_by = tuple(new_order_by)
+
+        # Clear SELECT clause as all annotation references were inlined by
+        # add_update_values() already.
+        query.clear_select_clause()
+        async with async_mark_for_rollback_on_error(using=self.db):
+            rows = await query.get_compiler(
```
Comment (Arfey) [inline django_async_backend/db/models/query.py]:
```python
rows = await query.get_compiler(self.db).execute_sql(ROW_COUNT)
```

**Review by Arfey — COMMENTED** [review]
The problem still exists. Right now, instead of using an async `SQLUpdateCompiler`, we pass the `connection=async_connections[self.db]` parameter. `SQLUpdateCompiler` differs from `SQLCompiler` in these methods:

- as_sql
- execute_sql
- execute_returning_sql
- pre_sql_setup

The main issue is `pre_sql_setup`. It's a sync method that calls another sync method, `as_sql`, which in turn calls `__str__` and triggers a SQL call for related updates. Not sure yet whether we can skip this temporarily. 

Next steps:

- Use `SQLUpdateCompiler` instead of the `connection=async_connections[self.db]` parameter.
- Investigate the related-updates problem (we can probably skip it temporarily, but we'll need to add an error message).

After review:

Add a lot of tests for update and `SQLUpdateCompiler` (I can help with this 😌).

File: codemon/config/db__models__query.yaml [hunk 3]
```diff
@@ -386,9 +382,6 @@ module:
             - func:
                 attr: execute_sql
               to_await: true
-            - func:
-                attr: get_compiler
-              replace_raw: 'query.get_compiler(connection=async_connections[using])'
```
Comment (bufke) [inline codemon/config/db__models__query.yaml:391]: Ah I see I forgot to remove this, thanks.
